### PR TITLE
ENH: Add a reader writer pair for HDF5 minute bar updates.

### PR DESCRIPTION
**Pull request by ehebert**
This format is intended for storing data for all sids of an asset type,
e.g. equities or futures for a session. bcolz is not used to avoid the overhead
of creating the directories and files for each asset (which numbers around ~8000
for active equities) can be removed since the update is meant to be read at
once, instead of supporting the random access pattern needed by the simulation.

This patch only adds the reader/writer pair, with the management of finding the
paths to delta files and the application of the updates to the bcolz write left
to internal loader code.

Also, the update reader interface is intentionally constrained to the data for
an entire session to allow for an implementation that allows for mid-session updates.

### CODE REVIEW

File: zipline/data/minute_bars.py [hunk 1]
```diff
@@ -1242,3 +1246,94 @@ def load_raw_arrays(self, fields, start_dt, end_dt, sids):
...
+        Write the frames to the target HDF5 file, using the format used by
+        ``pd.Panel.to_hdf``
+
+        Parameters
+        ----------
+        frames : iter[(int, DataFrame)] or dict[int -> DataFrame]
+            An iterable or other mapping of sid to the corresponding OHLCV
+            pricing data.
+        """
+        _store = HDFStore(
```
Comment (llllllllll) [inline zipline/data/minute_bars.py]: this can be used as a ctx manager

File: zipline/data/minute_bars.py [hunk 2]
```diff
@@ -1242,3 +1246,94 @@ def load_raw_arrays(self, fields, start_dt, end_dt, sids):
...
+            The minutes for which to read the pricing updates.
+        sids : iter[int]
+            The sids for which to read the pricing updates.
+
+        Returns
+        -------
+        data : iter[(int, DataFrame)]
+            Returns an iterable of ``sid`` to the corresponding OHLCV data.
+        """
+        raise NotImplementedError
```
Comment (llllllllll) [inline zipline/data/minute_bars.py]: raise an instance instead of the type

File: zipline/data/minute_bars.py [hunk 3]
```diff
@@ -1242,3 +1246,94 @@ def load_raw_arrays(self, fields, start_dt, end_dt, sids):
...
+class H5MinuteBarUpdateWriter(object):
+    """
+    Writer for files containing minute bar updates for consumption by a writer
+    for a ``MinuteBarReader`` format.
+
+    Parameters
+    ----------
+    path : str
+        The destination path.
+    complevel : int
```
Comment (llllllllll) [inline zipline/data/minute_bars.py]: `int, optional`

File: zipline/data/minute_bars.py [hunk 4]
```diff
@@ -1242,3 +1246,94 @@ def load_raw_arrays(self, fields, start_dt, end_dt, sids):
...
+            An iterable or other mapping of sid to the corresponding OHLCV
+            pricing data.
+        """
+        _store = HDFStore(
+            self._path, 'w', complevel=self._complevel, complib=self._complib)
+        panel = pd.Panel.from_dict(dict(frames))
+        panel.to_hdf(_store, 'updates')
+        _store.close()
+        h5file = tables.open_file(self._path, mode='r+')
+        metadata = h5file.create_group('/', 'metadata')
```
Comment (llllllllll) [inline zipline/data/minute_bars.py]: can you not write metadata with HDFStore?

File: tests/data/test_minute_bars.py [hunk 5]
```diff
@@ -1105,3 +1107,55 @@ def test_early_market_close(self):
...
+            },
+            index=minutes)
+
+        frames = {1: data_1, 2: data_2}
+        update_path = self.instance_tmpdir.getpath('updates.h5')
+        update_writer = H5MinuteBarUpdateWriter(update_path)
+        update_writer.write(frames)
+
+        update_reader = H5MinuteBarUpdateReader(update_path)
+        self.writer.write(update_reader.read(minutes, sids))
```
Comment (llllllllll) [inline tests/data/test_minute_bars.py]: should we add a helper somewhere like `apply_updates`?

**Comment by ehebert** [comment]
Just added `tables==3.3.0` to requirements to fix build issues.

File: zipline/data/minute_bars.py (hunk 3 above)
Comment (ehebert) [inline zipline/data/minute_bars.py]: Fixing.

File: zipline/data/minute_bars.py (hunk 2 above)
Comment (ehebert) [inline zipline/data/minute_bars.py]: Fixing.

File: zipline/data/minute_bars.py (hunk 4 above)
Comment (ehebert) [inline zipline/data/minute_bars.py]: Could not find a way via HDFStore to set it on the file.
However looked through the docs for `h5file.File` again and found `set_node_attr` so using this instead:
```
with tables.open_file(self._path, mode='r+') as h5file:
       h5file.set_node_attr('/', 'version', 0)
```

File: tests/data/test_minute_bars.py (hunk 5 above)
Comment (ehebert) [inline tests/data/test_minute_bars.py]: I am open to it. Ok with my following up with it after dust settles on the review of internal branch (where the internal branch may be changed to use `apply_updates`, with a corresponding PR adding that function.)

**Pull request merged**